release 0.7.17  优酷下载失败

annie https://v.youku.com/v_show/id_XMjI2ODA4MTQw.html
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x63da17]

goroutine 22 [running]:
github.com/iawia002/annie/request.Request(0x71eef5, 0x3, 0x0, 0x0, 0x0, 0x0, 0xc04232bd28, 0x0)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/request/request.go:97 +0xb47
github.com/iawia002/annie/downloader.writeFile(0x0, 0x0, 0xc04235c030, 0xc04232bd28, 0xc0420bf1e0, 0x0, 0x0, 0x0)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/downloader/downloader.go:91 +0x92
github.com/iawia002/annie/downloader.Save(0x0, 0x0, 0x2ee99a9, 0x0, 0x0, 0xc0420440c0, 0x56, 0xc04230d850, 0xf, 0xc0420bf1e0)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/downloader/downloader.go:181 +0x989
github.com/iawia002/annie/downloader.VideoData.Download.func1(0xc0420478e0, 0x0, 0x0, 0x2ee99a9, 0x0, 0x0, 0xc0420440c0, 0x56, 0xc04230d850, 0xf, ...)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/downloader/downloader.go:329 +0xba
created by github.com/iawia002/annie/downloader.VideoData.Download
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/downloader/downloader.go:327 +0x3b9
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x63da17]

goroutine 23 [running]:
github.com/iawia002/annie/request.Request(0x71eef5, 0x3, 0x0, 0x0, 0x0, 0x0, 0xc042325d28, 0x0)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/request/request.go:97 +0xb47
github.com/iawia002/annie/downloader.writeFile(0x0, 0x0, 0xc042004088, 0xc042325d28, 0xc0420bf1e0, 0x0, 0x0, 0x0)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/downloader/downloader.go:91 +0x92
github.com/iawia002/annie/downloader.Save(0x0, 0x0, 0x302f3f1, 0x0, 0x0, 0xc0420440c0, 0x56, 0xc04230d870, 0xf, 0xc0420bf1e0)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/downloader/downloader.go:181 +0x989
github.com/iawia002/annie/downloader.VideoData.Download.func1(0xc0420478e0, 0x0, 0x0, 0x302f3f1, 0x0, 0x0, 0xc0420440c0, 0x56, 0xc04230d870, 0xf, ...)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/downloader/downloader.go:329 +0xba
created by github.com/iawia002/annie/downloader.VideoData.Download
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/downloader/downloader.go:327 +0x3b9

--- a/README.md
+++ b/README.md
@@ -499,7 +499,7 @@ $ annie -j https://www.bilibili.com/video/av20203945
 
 ```
   -ccode string
-    	Youku ccode (default "010101500003")
+    	Youku ccode
 ```
 
 

--- a/extractors/youku_test.go
+++ b/extractors/youku_test.go
@@ -8,7 +8,7 @@ import (
 )
 
 func init() {
-	config.Ccode = "010101500003"
+	config.Ccode = "0515"
 }
 
 func TestYouku(t *testing.T) {
@@ -23,7 +23,7 @@ func TestYouku(t *testing.T) {
 				URL:     "http://v.youku.com/v_show/id_XMzUzMjE3NDczNg==.html",
 				Title:   "车事儿：智能汽车已经不在遥远 东风风光iX5发布",
 				Size:    22692900,
-				Quality: "mp4hd2 1280x720",
+				Quality: "mp4hd2v2 1280x720",
 			},
 		},
 		{
@@ -32,7 +32,7 @@ func TestYouku(t *testing.T) {
 				URL:     "http://v.youku.com/v_show/id_XMzQ1MTAzNjQwNA==.html",
 				Title:   "这！就是街舞 第一季 第3期：百强“互杀”队长不忍直视",
 				Size:    750911635,
-				Quality: "mp4hd2 1280x720 国语",
+				Quality: "mp4hd2v2 1280x720 国语",
 			},
 		},
 	}

--- a/main.go
+++ b/main.go
@@ -37,7 +37,7 @@ func init() {
 		"Playlist video items to download. Separated by commas like: 1,5,6",
 	)
 	flag.BoolVar(&config.Caption, "C", false, "Download captions")
-	flag.StringVar(&config.Ccode, "ccode", "010101500003", "Youku ccode")
+	flag.StringVar(&config.Ccode, "ccode", "0515", "Youku ccode")
 	flag.IntVar(
 		&config.RetryTimes, "retry", 100, "How many times to retry when the download failed",
 	)

--- a/request/request.go
+++ b/request/request.go
@@ -93,10 +93,15 @@ func Request(
 		if requestError == nil && res.StatusCode < 400 {
 			break
 		} else if i+1 >= config.RetryTimes {
-			log.Println(url)
-			panic(fmt.Sprintf(
-				"request error: %v HTTP %d", requestError, res.StatusCode,
-			))
+			var err string
+			if requestError != nil {
+				err = fmt.Sprintf("request error: %v", requestError)
+			} else {
+				err = fmt.Sprintf(
+					"%s request error: HTTP %d", url, res.StatusCode,
+				)
+			}
+			panic(err)
 		}
 		time.Sleep(1 * time.Second)
 	}

